src/components/Main.tsx

```diff
@@ -63,11 +66,14 @@ export function Main({
   /** Set of supported languages */
   supportedLanguages: ConsentManagerLanguageKey[];
 }): JSX.Element {
+  const { formatMessage } = useIntl();
   // need to focus the first button in the modal when the modal is opened
```

Comment:
I'm not sure I'd agree with this comment. 

If a dialog is opened by a manual trigger (like a button click) then I'm pretty sure focus should go to the close button, which is ideally also the first element in the dialog. 

If it was opened automatically (i.e. on page load), then I would worry that the user will not have context for where they are or what the button is for if they are immediately focused on a button. I'm honestly not sure what the solution should be, since I don't think setting `tabIndex` on the header element really makes sense either, and other than that, there isn't a way to start the user's screenreader at the top of the dialog. More research might have to be o

Reply:
Oh, maybe [this](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/document_role#:~:text=Adding%20role%3D%22document%22%20and%20tabindex%3D%220%22%20to%20the%20element%20containing%20the%20text%20within%20a%20widget%20enables%20the%20screen%20reader%20user%20to%20press%20the%20Tab%20key%20to%20place%20focus%20on%20the%20document%20element%20and%20read%20the%20text%20with%20the%20screen%20reader%27s%20reading%20cursor.) is the answer?

Reply:
ive tested this with a screenreader and on initial focus it reads out the dialog context that youre placed in, thats the reason im tagging an aria label/description on the dialog element. imo i think the user is given enough context here